handlers_test.go

```diff
@@ -57,3 +59,56 @@ func TestHandlers(t *testing.T) {
 	assert.Equal(t, h1, hmap.find(m1b))
 	assert.Equal(t, h2, hmap.find(m2b))
 }
+
+func procedure(svc, method string) string {
+	return fmt.Sprintf("%s::%s", svc, method)
+}
+
+func makeInboundCall(svc, method string, logger Logger) *InboundCall {
+	// need to populate connection.log to avoid nil pointer
+	conn := &Connection{}
+	// can't inline due to embeds
+	conn.log = logger
+	return &InboundCall{
+		serviceName:  svc,
+		method:       []byte(method),
+		methodString: method,
+		conn:         conn,
+	}
+}
+
+func TestUserHandlerWithIgnore(t *testing.T) {
+	const (
+		svc                    = "svc"
+		userHandleMethod       = "method"
+		userHandleIgnoreMethod = "ignoreMethod"
+		runs                   = 3
+	)
+
+	userCounter, channelCounter := map[string]int{}, map[string]int{}
+
+	opts := &ChannelOptions{
+		Handler:       recorderHandler(userCounter),
+		IgnoreMethods: []string{procedure(svc, userHandleIgnoreMethod)},
+	}
+	ch, err := NewChannel(svc, opts)
+	require.NoError(t, err, "error creating a TChannel channel")
+
+	// channel should be able to handle user ignored methods
+	ch.Register(recorderHandler(channelCounter), userHandleIgnoreMethod)
+
+	call, ignoreCall := makeInboundCall(svc, userHandleMethod, NullLogger), makeInboundCall(svc, userHandleIgnoreMethod, NullLogger)
+
```

Comment:
make a real call using the tchannel call using a client, avoid creating partial objects like connection / inbound call